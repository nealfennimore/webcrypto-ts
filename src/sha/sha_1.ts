import * as alg from "../alg";
import { WebCrypto } from "../crypto";
import { ShaShared } from "./shared";

export namespace SHA_1 {
    export const digest = (data: BufferSource) =>
        WebCrypto.digest(alg.SHA.Variant.SHA_1, data);

    export const hexify = ShaShared.hexify;
}
