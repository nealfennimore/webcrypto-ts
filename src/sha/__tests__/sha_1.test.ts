import { SHA_1 } from "../sha_1";

describe("SHA_1", () => {
    it("should hash", async () => {
        const data = encode("a message");
        const hash = await SHA_1.digest(data);
        expect(hash.byteLength).toEqual(20);
    });
    it("should hexify", async () => {
        const data = encode("a message");
        const hash = await SHA_1.digest(data);
        expect(SHA_1.hexify(hash)).toMatchSnapshot();
    });
});
