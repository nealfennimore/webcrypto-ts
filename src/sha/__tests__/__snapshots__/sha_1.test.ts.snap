// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHA_1 should hexify 1`] = `"00f2d6699acc796593a3cd60f5a039955e2d03c2"`;
